### DISCUSSION TITLE
Change the GPU in a single GPU Training

### DISCUSSION THREAD

**Discussion by mchlsdrv** [MIC-DKFZ/nnUNet#497] [Q&A] [answered]
Hi, I'm running on a cluster with four GPUs, which is being utilized by other users, and I'd like to select another GPU rather than the 0. I saw in the explanation that it is possible for the multi-GPU training, but how can I do it in a single-GPU training?

Thanks in advance,
Michael

**Comment by FabianIsensee** [comment, marked as answer]
Hi,
`CUDA_VISIBLE_DEVICES=X nnUNet_train 3d_fullres nnUNetTrainerV2 TASKID FOLD`
is what you should use.
Best,
Fabian

**Reply by mchlsdrv** [reply]
Hi, when I use this command (i.e., CUDA_VISIBLE_DEVICES=1 ...) it still uses the 'pin_memory' of GPU0, and results in memory overflow.  What could be the reason? 

Thanks

**Reply by FabianIsensee** [reply]
If you use this command it should not be using GPU0 at all! It will say it uses gpu0, but that it from the perspective of the program. The program only sees one GPU (which is GPU 1 in your case) and will assume that is gpu 0

**Reply by asjadnabeel** [reply]
@FabianIsensee Perfect answer, i was confused on the same, Thanks bro
